Plugin runtime error

Plugin config:
```xml
        <plugin>
          <groupId>org.llorllale</groupId>
          <artifactId>loggit-maven-plugin</artifactId>
          <version>0.8.0-rc6</version>
        </plugin>
```

Execution:
```
llorllale:~/dev/java/youtrack-api$ mvn -e loggit:changelog
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building youtrack-api 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- loggit-maven-plugin:0.8.0-rc6:changelog (default-cli) @ youtrack-api ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.345 s
[INFO] Finished at: 2018-03-11T10:38:46-04:00
[INFO] Final Memory: 17M/197M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.llorllale:loggit-maven-plugin:0.8.0-rc6:changelog (default-cli) on project youtrack-api: Execution default-cli of goal org.llorllale:loggit-maven-plugin:0.8.0-rc6:changelog failed: Command org.eclipse.jgit.api.LogCommand was called in the wrong state -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.llorllale:loggit-maven-plugin:0.8.0-rc6:changelog (default-cli) on project youtrack-api: Execution default-cli of goal org.llorllale:loggit-maven-plugin:0.8.0-rc6:changelog failed: Command org.eclipse.jgit.api.LogCommand was called in the wrong state
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.llorllale:loggit-maven-plugin:0.8.0-rc6:changelog failed: Command org.eclipse.jgit.api.LogCommand was called in the wrong state
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
        ... 20 more
Caused by: java.lang.IllegalStateException: Command org.eclipse.jgit.api.LogCommand was called in the wrong state
        at org.eclipse.jgit.api.GitCommand.checkCallable(GitCommand.java:123)
        at org.eclipse.jgit.api.LogCommand.add(LogCommand.java:347)
        at org.eclipse.jgit.api.LogCommand.add(LogCommand.java:202)
        at org.llorllale.mvn.plgn.loggit.TagsOf.lambda$null$2(TagsOf.java:67)
        at org.cactoos.func.UncheckedFunc.lambda$apply$0(UncheckedFunc.java:58)
        at org.cactoos.func.IoCheckedBiFunc.apply(IoCheckedBiFunc.java:66)
        at org.cactoos.func.UncheckedBiFunc.apply(UncheckedBiFunc.java:60)
        at org.cactoos.func.UncheckedFunc.apply(UncheckedFunc.java:59)
        at org.cactoos.iterator.Mapped.next(Mapped.java:70)
        at org.cactoos.iterator.Mapped.next(Mapped.java:70)
        at org.cactoos.scalar.And.value(And.java:166)
        at org.llorllale.mvn.plgn.loggit.TagsOf.lambda$iterator$4(TagsOf.java:75)
        at org.cactoos.scalar.UncheckedScalar.lambda$value$0(UncheckedScalar.java:58)
        at org.cactoos.func.UncheckedFunc.lambda$apply$0(UncheckedFunc.java:58)
        at org.cactoos.func.IoCheckedBiFunc.apply(IoCheckedBiFunc.java:66)
        at org.cactoos.func.UncheckedBiFunc.apply(UncheckedBiFunc.java:60)
        at org.cactoos.func.UncheckedFunc.apply(UncheckedFunc.java:59)
        at org.cactoos.scalar.UncheckedScalar.value(UncheckedScalar.java:59)
        at org.llorllale.mvn.plgn.loggit.TagsOf.iterator(TagsOf.java:78)
        at org.xembly.Directives.append(Directives.java:250)
        at org.llorllale.mvn.plgn.loggit.DefaultCommit.asXml(DefaultCommit.java:67)
        at org.llorllale.mvn.plgn.loggit.DefaultLog.lambda$asXml$1(DefaultLog.java:76)
        at java.lang.Iterable.forEach(Iterable.java:75)
        at org.llorllale.mvn.plgn.loggit.DefaultLog.asXml(DefaultLog.java:75)
        at org.llorllale.mvn.plgn.loggit.Changelog.execute(Changelog.java:250)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        ... 21 more
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

--- a/src/main/java/org/llorllale/mvn/plgn/loggit/Changelog.java
+++ b/src/main/java/org/llorllale/mvn/plgn/loggit/Changelog.java
@@ -19,6 +19,7 @@ package org.llorllale.mvn.plgn.loggit;
 import com.jcabi.xml.XML;
 import java.io.File;
 import java.io.IOException;
+import java.util.Optional;
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
@@ -291,8 +292,8 @@ public final class Changelog extends AbstractMojo {
    */
   private XML preprocess(XML xml) throws IOException {
     return new Pattern(this.includeRegex).transform(
-      new EndTag(this.endTag).transform(
-        new StartTag(this.startTag).transform(
+      new EndTag(Optional.ofNullable(this.endTag).orElse("")).transform(
+        new StartTag(Optional.ofNullable(this.startTag).orElse("")).transform(
           new Limit(this.maxEntries).transform(xml)
         )
       )

--- a/src/main/java/org/llorllale/mvn/plgn/loggit/TagsOf.java
+++ b/src/main/java/org/llorllale/mvn/plgn/loggit/TagsOf.java
@@ -17,11 +17,8 @@
 package org.llorllale.mvn.plgn.loggit;
 
 import java.util.Iterator;
-import org.cactoos.iterable.Mapped;
-import org.cactoos.iterator.Filtered;
-import org.cactoos.scalar.And;
 import org.cactoos.scalar.UncheckedScalar;
-import org.eclipse.jgit.api.LogCommand;
+import org.eclipse.jgit.lib.Ref;
 import org.eclipse.jgit.lib.Repository;
 import org.eclipse.jgit.revwalk.RevCommit;
 import org.xembly.Directive;
@@ -54,25 +51,12 @@ final class TagsOf implements Iterable<Directive> {
     return new UncheckedScalar<>(
       () -> {
         final Directives dirs = new Directives();
-        final LogCommand log = new org.eclipse.jgit.api.Git(this.repo).log();
-        new And(
-          tag -> {
-            if (new Filtered<>(rev -> rev.equals(this.commit), log.call().iterator()).hasNext()) {
-              dirs.add("tag").set(tag.getName().split("/")[2]).up();
-            }
-            return true;
-          },
-          new Mapped<>(
-            peeled -> {
-              log.add(peeled.getPeeledObjectId());
-              return peeled;
-            },
-            new Mapped<>(
-              ref -> this.repo.peel(ref),
-              new org.eclipse.jgit.api.Git(this.repo).tagList().call()
-            )
-          )
-        ).value();
+        for (Ref tag : new org.eclipse.jgit.api.Git(this.repo).tagList().call()) {
+          final Ref peeled = this.repo.peel(tag);
+          if (this.commit.getId().equals(peeled.getPeeledObjectId())) {
+            dirs.add("tag").set(tag.getName().split("/")[2]).up();
+          }
+        }
         return dirs.iterator();
       }
     ).value();

